hazelcast/src/test/java/com/hazelcast/osgi/HazelcastOSGiServiceTest.java

```diff
@@ -24,17 +28,86 @@
 import static org.junit.Assert.assertFalse;
 import static org.junit.Assert.assertNull;
 import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
 
 @RunWith(HazelcastSerialClassRunner.class)
 @Category(QuickTest.class)
 public class HazelcastOSGiServiceTest {
 
+    private static String loggingType;
+    private static Level log4jLogLevel;
+
     private TestBundle bundle;
     private TestBundleContext bundleContext;
+    private TestBundleRegisterDeregisterListener registerDeregisterListener;
+
+    private class TestBundleRegisterDeregisterListener implements TestBundle.RegisterDeregisterListener {
+
+        final String className;
+
+        boolean throwExceptionOnRegister;
+        boolean throwExceptionOnDeregister;
+
+        private TestBundleRegisterDeregisterListener() {
+            this.className = null;
+        }
+
+        private TestBundleRegisterDeregisterListener(String className) {
+            this.className = className;
+        }
+
+        @Override
+        public void onRegister(String clazz, TestServiceReference serviceReference) {
+            if (throwExceptionOnRegister) {
+                if (className != null) {
+                    if (clazz.equals(className)) {
+                        throw new IllegalStateException("You cannot register!");
+                    }
+                } else {
+                    throw new IllegalStateException("You cannot register!");
+                }
+            }
+        }
+
+        @Override
+        public void onDeregister(String clazz, TestServiceReference serviceReference) {
+            if (throwExceptionOnDeregister) {
+                if (className != null) {
+                    if (clazz.equals(className)) {
+                        throw new IllegalStateException("You cannot deregister!");
+                    }
+                } else {
+                    throw new IllegalStateException("You cannot deregister!");
+                }
+            }
+        }
+    }
+
+    @BeforeClass
+    public static void beforeClass() {
+        loggingType = System.getProperty("hazelcast.logging.type");
+        Logger rootLogger = Logger.getRootLogger();
+        if ("log4j".equals(loggingType)) {
+            log4jLogLevel = rootLogger.getLevel();
+        }
+        rootLogger.setLevel(Level.TRACE);
+        System.setProperty("hazelcast.logging.type", "log4j");
```

Comment:
Do we need this logging hacks in `@AfterClass` and `@BeforeClass`? Did you use it only while developing test cases? if so, we are safe to remove them...